cmd/root.go

```diff
@@ -239,8 +244,27 @@ func parseConfig(cmd *cobra.Command, conf *proxy.Config, args []string) error {
 					return newBadCommandError(fmt.Sprintf("unix query param should be only one value: %q", a))
 				}
 				ic.UnixSocket = u[0]
+			}
 
+			if iam, ok := q["auto-iam-authn"]; ok {
+				if len(iam) != 1 {
+					return newBadCommandError(fmt.Sprintf("auto iam authn param should be only one value: %q", iam))
+				}
+				enableIAMAuthN := true
+				disableIAMAuthn := false
+				switch iam[0] {
```

Comment:
semi-nit: Do we care about case sensitivity?

Reply:
I think we do -- but right now all our query params are case sensitive. So I'm inclined to address this problem in one go.